### PR TITLE
kubectl apply add --overwrite flag

### LINKED ISSUE #17238: kubectl apply --overwrite
This issue tracks the implementation of the --overwrite flag in kubectl apply, per [this comment](https://github.com/kubernetes/kubernetes/pull/16980#issuecomment-156101513) in #16980. Fixes #13576.

### PR DESCRIPTION
**Pull request by adohe-zz**
``` release-note
Add "kubectl --overwrite" flag to automatically resolve conflicts between the modified and live configuration using values from the modified configuration.
```

fixes #17238

This PR just add the `--overwrite` flag, pkg/strategicpatch has already support overwrite. @bgrant0607 @jackgr ptal.

### CODE REVIEW

**Comment by bgrant0607** [comment]
@AdoHe Thanks, though this needs to wait until after the 1.3 code freeze ends.

**Comment by bgrant0607** [comment]
cc @kubernetes/kubectl

**Comment by AdoHe** [comment]
@bgrant0607 it's ok. I just get more time to do kubernete related work and will keep going.

**Comment by pwittrock** [comment]
@johndmulhausen What is the process for updating the docs repo with changes from generated files?  Is there a script for this?

File: pkg/kubectl/cmd/apply.go [hunk 1]
```diff
@@ -69,6 +69,7 @@ func NewCmdApply(f *cmdutil.Factory, out io.Writer) *cobra.Command {
 	usage := "Filename, directory, or URL to file that contains the configuration to apply"
 	kubectl.AddJsonFilenameFlag(cmd, &options.Filenames, usage)
 	cmd.MarkFlagRequired("filename")
+	cmd.Flags().Bool("overwrite", true, "Ignore conflicts between modified and current configuration")
```
Comment (pwittrock) [inline pkg/kubectl/cmd/apply.go]: Please clarify what ignoring means in this help text.  Maybe `Automatically resolve conflicts between the modified and live configurations by using values from the modified configuration.`

**Comment by pwittrock** [comment]
@k8s-bot test this issue: #IGNORE

Tests hung

**Comment by pwittrock** [comment]
@k8s-bot test this issue: #IGNORE

Some infrastructure failure.  Lots of tests broken without explanation.

**Comment by AdoHe** [comment]
@pwittrock I just rebase and update this, ptal.

**Comment by pwittrock** [comment]
@AdoHe Thanks Tony

**Pull request merged**